### PR TITLE
SIMSBIOHUB-108-2: Secure Documents and See Document Status When Security Rules Applied

### PR DESCRIPTION
**Pull request by KjartanE**
# Overview

## Links to Jira tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-108

## Description of relevant changes

- {List all relevant changes, in particular anything that will help the reviewers test/verify this PR}

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/src/services/artifact-service.ts [hunk 1]
```diff
@@ -110,7 +110,27 @@ export class ArtifactService extends DBService {
     return this.artifactRepository.getArtifactsByDatasetId(datasetId);
   }
 
+  /**
+   * Retrieves an artifact by its primary key.
+   *
+   * @param {number} artifactId
+   * @return {*}  {Promise<Artifact>}
+   * @memberof ArtifactService
+   */
```
Comment (curtisupshall) [inline api/src/services/artifact-service.ts:119]: +1 for added documentation 👍

File: app/src/features/datasets/components/AttachmentItemMenuButton.tsx [hunk 2]
```diff
@@ -50,7 +58,7 @@ const AttachmentItemMenuButton: React.FC<IAttachmentItemMenuButtonProps> = (prop
               'aria-labelledby': 'basic-button'
             }}>
             <MenuItem
-              disabled={!props.hasAdministrativePermissions}
+              disabled={checkPermissions()}
```
Comment (curtisupshall) [inline app/src/features/datasets/components/AttachmentItemMenuButton.tsx]: Could avoid an unnecessary function call by just using an expression in here:

```jsx
<MenuItem
  disabled={props.hasAdministrativePermissions
    ? false
    : props.artifact.supplementaryData.persecutionAndHarm !== SECURITY_APPLIED_STATUS.UNSECURED
  }
  ...
>

File: app/src/features/datasets/components/DatasetArtifacts.tsx [hunk 3]
```diff
@@ -61,8 +61,13 @@ const DatasetAttachments: React.FC<IDatasetAttachmentsProps> = (props) => {
...
   artifactsDataLoader.load();
 
-  const artifactsList = artifactsDataLoader.data?.artifacts || [];
-  const numPendingDocuments = artifactsList.filter((artifact) => artifact.security_review_timestamp === null).length;
+  const artifactsList = artifactsDataLoader.data || [];
+
+  const numPendingDocuments =
+    artifactsList.filter(
+      (artifact) => artifact.supplementaryData.persecutionAndHarm === SECURITY_APPLIED_STATUS.PENDING
+    ).length || 0;
```
Comment (curtisupshall) [inline app/src/features/datasets/components/DatasetArtifacts.tsx]: You can remove this coelesce; `artifactsList` is guarenteed to be an array, so the length is guarenteed to be defined.

File: app/src/features/datasets/components/DatasetArtifacts.tsx [hunk 4]
```diff
@@ -145,7 +158,7 @@ const DatasetAttachments: React.FC<IDatasetAttachmentsProps> = (props) => {
             color="primary"
             startIcon={<Icon path={mdiLockPlus} size={1} />}
             onClick={() => setOpenApplySecurity(true)}
-            disabled={selectedArtifacts.length > 0 ? false : true}>
+            disabled={hasAdministrativePermissions && selectedArtifacts.length > 0 ? false : true}>
```
Comment (curtisupshall) [inline app/src/features/datasets/components/DatasetArtifacts.tsx]: Nit: Can be simplified to `disabled={!hasAdministrativePermissions || selectedArtifacts.length === 0}`

**Review by curtisupshall — DISMISSED** [review]
Code looks great!

**Review by al-rosenthal — APPROVED** [review]
🕶️

**Review by curtisupshall — APPROVED** [review]

**Pull request merged**